java/core/src/java/org/apache/orc/impl/WriterImpl.java

```diff
@@ -1018,4 +1005,20 @@ private WriterEncryptionVariant[] setupEncryption(
   public long estimateMemory() {
     return this.treeWriter.estimateMemory();
   }
+
+  /**
+   * Get next chunk size if row selected enable.
+   * We will find the longest size of chunk that will be less than maxSize.
+   * @param posn the position where we are
+   * @param maxSize the largest chunk size in this selection
+   */
+  private int nextSelectedRowSize(int posn, int maxSize, int[] selected) {
```

Comment:
Good refactoring. The only concern is that before the change we know posn and (posn + len) will never exceed selected.length. The new method has potential misuse if someone else has passed wrong maxSize when calling it.

Reply:
Good catching! Will add an assertion here.